bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/ThingHandlerKeypad.java

```diff
@@ -26,25 +38,84 @@
  */
 @NonNullByDefault
 public class ThingHandlerKeypad extends CaddxBaseThingHandler {
-    /**
-     * Constructor.
-     *
-     * @param thing
-     */
+    private final Logger logger = LoggerFactory.getLogger(ThingHandlerKeypad.class);
+
     public ThingHandlerKeypad(Thing thing) {
         super(thing, CaddxThingType.KEYPAD);
     }
 
     @Override
     public void updateChannel(ChannelUID channelUID, String data) {
+        if (channelUID.getId().equals(CaddxBindingConstants.KEYPAD_KEY_PRESSED)) {
+            StringType stringType = new StringType(data);
+            updateState(channelUID, stringType);
+        }
     }
 
     @Override
-    public void handleCommand(ChannelUID channelUID, Command command) {
+    public void caddxEventReceived(CaddxEvent event, Thing thing) {
+        logger.trace("caddxEventReceived(): Event Received - {}.", event);
+
+        if (getThing().equals(thing)) {
+            CaddxMessage message = event.getCaddxMessage();
+            CaddxMessageType mt = message.getCaddxMessageType();
+            ChannelUID channelUID = null;
```

Comment:
I see no reason that this variable needs to be defined outside of the for loop.

Reply:
done